src/components/pages/dashboard-pages/EditProfile.jsx

```diff
@@ -1,20 +1,132 @@
+import {
+  getUserData,
+  updateUser,
+  updateUserPhoto,
+} from '../../../util/DataBaseRequests';
+
 import EditProfileForm from './EditProfileForm';
 import subjectOptions from '../../../data/subjects';
 import { useAuth } from '../../../AuthProvider';
 import { useNavigate } from 'react-router-dom';
+import { useState } from 'react';
 
 const EditProfile = () => {
-  const { userData } = useAuth();
+  const { userData, setUserData } = useAuth();
+  const [category, setCategory] = useState(null); //state for select
+  const [formData, setFormData] = useState({
+    firstName: userData.firstName,
+    lastName: userData.lastName,
+    email: userData.email,
+    education: userData.education,
+    experience: userData.experience,
+    subjectArea: userData.subjectArea || category,
+    aboutMe: userData.aboutMe,
+  });
+  const [file, setFile] = useState();
+  const [formErrors, setFormErrors] = useState({});
+  const [isUpload, setIsUpload] = useState(false);
   const navigate = useNavigate();
   const options = subjectOptions;
 
+  const handleChange = (e) => {
+    setFormData({ ...formData, [e.target.name]: e.target.value });
+    setFormErrors({ ...formErrors, [e.target.name]: '', form: '' });
+  };
+  
+  const setSubjects = (value) => {
```

Comment:
Let's follow common naming conventions and reserve the 'set' prefix for state setter functions.